Source/Core/VideoCommon/RenderBase.cpp

```diff
@@ -476,6 +476,23 @@ float Renderer::CalculateDrawAspectRatio(int target_width, int target_height)
   }
 }
 
+std::tuple<float, float> Renderer::ScaleToDisplayAspectRatio(const int width, const int height)
+{
+  // Scale either the width or height depending the content aspect ratio.
+  // This way we preserve as much resolution as possible when scaling.
+  float ratio = CalculateDrawAspectRatio(width, height);
+  if (ratio >= 1.0f)
+  {
+    // Preserve horizontal resolution, scale vertically.
+    return std::make_tuple(static_cast<float>(width), static_cast<float>(height) * ratio);
+  }
+  else
```

Comment:
`else` isn't necessary, since the above body terminates with a `return`, other than that, it looks fine.